easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb

```diff
@@ -0,0 +1,247 @@
+easyblock = 'PythonBundle'
+
+name = 'pyiron'
+version = '0.5.1'
+
+homepage = 'https://github.com/pyiron/pyiron'
+description = "An integrated development environment (IDE) for computational materials science."
+
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('matplotlib', '3.7.2'),
+    ('h5py', '3.9.0'),
+    ('PyYAML', '6.0'),
+    ('phonopy', '2.20.0'),
+    ('spglib-python', '2.1.0'),
+    ('ASE', '3.22.1'),
+    ('PyTables', '3.8.0'),
+    ('nglview', '3.0.8'),
+    ('tqdm', '4.66.1'),
+    ('molmod', '1.4.8'),
+    ('scikit-learn', '1.3.1'),
+    ('yaff', '1.6.0'),
+    ('TAMkin', '1.2.6'),
+    ('QuickFF', '2.2.7'),
+    ('bcrypt', '4.0.1'),
+    ('ruamel.yaml', '0.17.32'),
+    ('plotly.py', '5.16.0'),
+    ('Flask', '2.3.3'),
+    ('boto3', '1.28.70'),
+    ('sympy', '1.12'),
+    ('maturin', '1.1.0'),
+    ('poetry', '1.5.1'),
+]
+
+check_ldshared = True
+use_pip = True
+
+exts_list = [
+    ('pint', '0.23', {
+        'source_tmpl': 'Pint-%(version)s.tar.gz',
+        'checksums': ['e1509b91606dbc52527c600a4ef74ffac12fff70688aff20e9072409346ec9b4'],
+    }),
+    ('pydantic_core', '2.14.6', {
+        'source_tmpl': '%(name)s-%(version)s-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl',
```

Comment:
This is an issue as it won't work for non-x86 platforms. We have `pydantic` ECs for older toolchains. I updated them to 2023a and will again submit a PR to yours once that is tested.   
Similarly we have ECs for some of the extensions used, such as Greenlet and GitPython. They will be included in the PR I'm preparing. This will avoid rebuilding python packages used by other modules etc.

Reply:
Really nice work, thank you so much. I had PyDantic at least originally as a separate EC, but I couldn't get Rust to build it properly (some Rust version issue AFAIK).